### ISSUE TITLE
RKE loading wrong ingress template during cluster provisioning

### ISSUE THREAD

**Issue by ryansann** [rancher/rancher#26752]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1. In Rancher UI, navigate to settings
2. Scroll down to the `rke-metadata-config` section
3. Modify the config to use the following:
```json
{
  "refresh-interval-minutes": "1",
  "url": "https://raw.githubusercontent.com/ryansann/kontainer-driver-metadata/fa58036b8a5c7a2496444d4ac93e1983d86a0d91/data/data.json"
}
```
4. Add a cluster with RKE (I am using Digital Ocean though this isn't specific to a cloud provider)
5. Choose your node template
6. In Kubernetes Options, for the Kubernetes Version, choose: `v1.16.8-rancher1-4`
7. Create Cluster

**Result:**
The ingress daemon set is provisioned using the wrong template

**Other details that may be helpful:**
I have added a new ingress template for an updated ingress container, (see [data.json](https://raw.githubusercontent.com/ryansann/kontainer-driver-metadata/fa58036b8a5c7a2496444d4ac93e1983d86a0d91/data/data.json)). The new template is expected to be used based on the configuration in the rke-metadata-config, but a different template is being used. 

The way you can distinguish that the correct template is being used is by running the following command: `kubectl get configmap rke-ingress-controller -n kube-system -o yaml | grep runAsUser` against the cluster that RKE provisioned. The output should show a value of `101` for the runAsUser field. If the value is `33`, the wrong template is being used.

Per @kinarashah "it's a regression in 2.4 when kdm logic went through a refactor"

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.4.2
- Installation option (single install/HA): single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Infrastructure Provider
- Machine type (cloud/VM/metal) and specifications (CPU/memory): Cloud (Digital Ocean Droplet) - 2 vCPUs / 4GB RAM
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.8", GitCommit:"ec6eb119b81be488b030e849b9e64fda4caaf33c", GitTreeState:"clean", BuildDate:"2020-03-12T20:52:22Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b7f0
 Built:             Wed Mar 11 01:25:46 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b7f0
  Built:            Wed Mar 11 01:24:19 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Comment by sangeethah** [comment]
Related to https://github.com/rancher/rancher/issues/27350

**Comment by rancher-max** [comment]
#### Tested using a custom rancher image where it was not vendored in

- When pointing to the correct KDM branch (`dev-v2.4` in this case), all new kubernetes versions are available and come up successfully (`v1.17.6-rancher2-1`, `v1.16.10-rancher2-1`, `1.15.12-rancher2-2`)
- nginx-ingress-controller comes up successfully using image `rancher/nginx-ingress-controller:nginx-0.32.0-rancher1`
- All other system images come up successfully and the cluster is operational
